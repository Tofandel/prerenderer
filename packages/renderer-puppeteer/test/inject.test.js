const path = require('path')
const Prerenderer = require('@prerenderer/prerenderer')
const Renderer = require('@prerenderer/renderer-puppeteer')

const testDir = path.resolve(__dirname, '../../../tests/inject')

test("injects {foo: 'bar'} into window['__PRERENDER_INJECTED']", async () => {
  const prerenderer = new Prerenderer({
    staticDir: testDir,
    indexPath: 'inject-basic.html',
    renderer: new Renderer({
      inject: {
        foo: 'bar',
      },
    }),
  })

  await prerenderer.initialize()
  const renderedRoutes = await prerenderer.renderRoutes(['/'])
  await prerenderer.destroy()
  expect(renderedRoutes).toMatchSnapshot()
})

test("injects {foo: 'bar'} into window['__CUSTOM_INJECTED']", async () => {
  const prerenderer = new Prerenderer({
    staticDir: testDir,
    indexPath: 'inject-change-property.html',
    renderer: new Renderer({
      injectProperty: '__CUSTOM_INJECTED',
      inject: {
        foo: 'bar',
      },
    }),
  })

  await prerenderer.initialize()
  const renderedRoutes = await prerenderer.renderRoutes(['/'])
  await prerenderer.destroy()
  expect(renderedRoutes).toMatchSnapshot()
})
