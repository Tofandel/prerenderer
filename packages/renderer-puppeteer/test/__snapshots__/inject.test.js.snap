// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`injects {foo: 'bar'} into window['__CUSTOM_INJECTED'] 1`] = `
[
  {
    "html": "<!DOCTYPE html><html><head>
  <title>Prerenderer Test: Injection with custom name</title>
</head>
<body>
  <script>
    document.addEventListener('DOMContentLoaded', () => {
      document.body.innerHTML += \`<p>\${JSON.stringify(window['__CUSTOM_INJECTED'])}</p>\`
    })
  </script>


<p>{"foo":"bar"}</p></body></html>",
    "originalRoute": "/",
    "route": "/",
  },
]
`;

exports[`injects {foo: 'bar'} into window['__PRERENDER_INJECTED'] 1`] = `
[
  {
    "html": "<!DOCTYPE html><html><head>
  <title>Prerenderer Test: Basic Injection</title>
</head>
<body>
  <script>
    document.addEventListener('DOMContentLoaded', () => {
      document.body.innerHTML += \`<p>\${JSON.stringify(window['__PRERENDER_INJECTED'])}</p>\`
    })
  </script>


<p>{"foo":"bar"}</p></body></html>",
    "originalRoute": "/",
    "route": "/",
  },
]
`;
